### DISCUSSION TITLE
We Want Your Feedback For Appwrite 1.0!

### DISCUSSION THREAD

**Discussion by Haimantika** [appwrite/appwrite#3915] [General]
👋 Hello Appwriters!

We've recently released Appwrite 1.0 🥳 , which is our first release intended to be production ready. 

We're looking for feedback on how we did for this release. What did you like? What didn't work out for you? What can we do better in future releases? 

Leave your thoughts in a comment and upvote other comments that you agree with. 

This is a huge milestone for us and one of our biggest releases yet. We want to make sure we got it right and your feedback and guidance are indispensable for making it happen. We'll be combing through all your comments!

- Team Appwrite

**Comment by MrKrishnaAgarwal** [comment]
I've been using Appwrite's Services for more than 1 year and it's been a awesome experience with it and I was amazed to see Appwrite's Community which is so helpful not only related to Appwrite but Tech and Open Source Help.
It has been a very useful tool for me and I also participated in a Appwrite Hackathon earlier. 

Its very easy to use and The dashboard also. 

Kudos and Cheers to @eldadfux and whole Appwrite Team for making an amazing product and Congrats for being **#1 on Product Hunt for Product of the Day and #2 for Product of the week.**

FunFact: I got my PR merged recently in Appwrite (#3895).

**Reply by eldadfux** [reply]
Thank you so much! Congrats on your first PR, It's always the one you remember the most! ❤️

**Reply by MrKrishnaAgarwal** [reply]
> It's always the one you remember the most! ❤️

Yes, Definitely 💯

**Comment by rodgav** [comment]
These last two years have been very exciting to be using and/or testing Appwrite, always looking forward to a new release to see its new features or integrations and today to see a stable version 1.0.

What I like most about appwrite is that you can self-host it wherever you want: VPS or on your own computer.
What I use the most, how to say that I use more or use less, if I get to use everything from auth, database, realtime and storage.

To culminate thank and congratulate the entire appwrite community, which is always there to solve the thousands of questions that are made in github and discord.

Send a big hello to @eldadfux  for recommending me to use appwrite and leave firebase in a flutter facebook group.

**Reply by eldadfux** [reply]
Thank you so much! Happy to see you're having fun with Appwrite!

**Comment by Souvik2024** [comment]
Firstly Kudos to the Appwrite community on GitHub, Discord, and Twitter they are so kind & always helpful! Congratulations on the launch to @eldadfux and the team !! 

Talking about Appwrite, Functions are my favorite feature, and I heard that they are bringing some awesome improvements to functions very soon I hope we'll love them. 
This could make things a whole lot easier. Excited to try this out! The idea behind Appwrite is to make software development fun and simple 🤩 that's what I like the most .

Lastly Many congratulations again to  Appwrite team for the Product Hunt launch. It's great to see such a powerful open-source product. Indeed, it is a great choice for the backend. ❤️

Thank you !!!

**Reply by eldadfux** [reply]
Thank you so much!

**Comment by hiteshgarg123** [comment]
Firstly congratulations to team Appwrite on 1.0 release. I have seen very few communities which are super kind and supportive and appwrite is one of them.🚀

They offer support to anyone seeking help or guidance on their Discord. Even if the question is not related to appwrite itself and is more general framework/language based, they are always polite and humble to everyone and try to help people. 
It's really good too see that we have such great communities around.🙌

They also encourage everyone to contribute to open source, which is a great thing. Infact the whole Appwrite project itself is open source so people can (and should) really contribute to building this amazing tool and this great community.🙏

Again, Very congratulations to team Appwrite on this big Milestone ❤️

**Reply by eldadfux** [reply]
Thank you so much for all the kind words!

**Reply by Haimantika** [reply]
Thank you Hitesh! Means a lot!

**Comment by Mazahir26** [comment]
Hey, I came across appwrite a few months ago and I was just amazed. Finally a good firebase alternative (Supabase was good but it didn't have cloud functions back then).
There are soo many good things about appwrite.  I have been using appwrite to develop an project( its been an amazing experience, the discord community is very active and helpful especially @stnguyen90 ).

I have a few suggestions (Not complains just suggestions), 

- As you use maria DB under the hood, it would be great if you could expose relations between documents. You could give user a choice which would be revolutionary.
-  Comparing, Firebase client sdk and appwrite client sdk. I feel firebase's sdk have a lot more features to offer than appwrite. 

Lastly, I can't wait for appwrite cloud. Its gonna be amazing.

Thank You.

**Reply by eldadfux** [reply]
Hey @Mazahir26 thanks for the kind word and awesome feedback! These are exactly the kind of insights that help us make the product better!

Regarding relations, this is on the roadmap! Actually, new queries that we just introduced are designed to help us get there!

What SDK features are you missing compared to Firebase? I'd love to learn more about it.

Thanks again for the great feedback!

**Reply by Mazahir26** [reply]
Well, I might be nitpicking. But sometimes small features make a big difference.

- The SDK lack listeners, for example Auth Listener if the user logs in or logs out it triggers a function.
- Firebase caches everything (Appwrite SDK do caches Images though).

I hope Appwrite SDK's will grow and be much better over time(even better than firebase).
Thank you.😁

**Comment by sarkardeep** [comment]
Huge shoutout to @eldadfux , @Haimantika , @adityaoberai, and the entire Appwrite team for this amazing milestone. 

In Version 1.0.0, There are so many Features have been added, Bugs have been fixed, also some breaking changes are there.The recent versions of Appwrite 1.0 are phenomenal! Can't believe how much you guys spent countless hours ⏱️ perfecting the Appwrite architecture, stack, and API 👨‍💻 for making this product better with every update, and it shows in the quality that comes out. Keep up your outstanding efforts! It's not surprising that Appwrite is so popular among refined users.

Finally, I can't wait to see the appwrite cloud.

Lastly, Thanks to @stnguyen90 for always being there in discord to answer all doubts.

Love everything about the product and community 🎉💜💯 , Keep Growing !!!

**Reply by eldadfux** [reply]
Thank you so much @sarkardeep this means the world to us!

**Reply by Haimantika** [reply]
Thank you @sarkardeep for your kind words ☺️

**Comment by mazieri** [comment]
To start, I would like to congratulations all the Appwrite team. The Appwrite is really amazing. I using In my personal projects and is really good.

- I like to suggest the remote config to Appwrite, similar to Firebase Remote Config... If can add Analytics to, I really would love them so much

**Reply by Haimantika** [reply]
Thank you for your feedback! Remote config is something we have planned to work on! 😊

**Comment by noga-dev** [comment]
In order to truly contest Firebase it's necessary to have or at least streamline the majority if not all of its features. To that end:

1. Offline cache sync (like Firestore).
2. Push notifications (use Firebase under the hood).
3. Simplify functions (like invoking Firebase's is simple, but not Appwrite's).

And merge my PR since early July.... (https://github.com/appwrite/functions-starter/pull/31)

**Comment by anomic30** [comment]
Firstly huge congratulations to team Appwrite on the v1.0 release and the Product Hunt launch🥳. 
I came across Appwrite, a few months ago during the Appwrite **x** Dev.to hackathon. Up until that point I have been using Firebase. Then I found this awesome open-source alternative and I was blown away by its feature and potential. I had a super fun experience creating [Snake Arcade](https://dev.to/anomic30/retro-themed-snake-game-with-live-highscore-board-2lei) using Appwrite's database and real-time feature. The docs and the Discord community always helped me whenever I got stuck. 

Also, I had my first PR merged [#418](https://github.com/appwrite/awesome-appwrite/pull/418)

Lastly, I want to again congratulate the entire Appwrite team and the community💗
Super excited for the Appwrite Cloud 🚀🚀🚀

**Comment by brandonapt** [comment]
I started using Appwrite just about a month ago, and it made me completely relook my backend services. Appwrite was the first OSS replacement I used in my backend infrastructure (replacing firebase functions). It blew me away. Appwrite has made me look more into OSS replacements for proprietary software.

Excited for the future!

**Comment by donno2048** [comment]
Loved it! The first good open-source alternative for firebase, I just wish it had all the features it has...

**Comment by Aunali321** [comment]
I love it! it's the best firebase alternative out there. So easy and intuitive.
Would love it even more if it had more database options (Maybe MongoDB?).
Thank you for your work team!

**Comment by avifatal** [comment]
Hi,
First of all it feels that a lot of thinking is behind this project. small things like the docker simplicity to large things like permissions.
As a nodejs developer and a big hater of PHP, the technology is really transparent and allow me to enjoy nodejs without thinking about PHP and all of that stack.

The support in Discord is the best I have seen so far, including big and small brands. the best!!! and it is really important for adoption.

The only thing I have problem with, is that it is not so server-side friendly. We are missing features that are available in the client side.
such as real server-side impersonation without having the JWT key passed from the client.
So when you want to expose an API  server-to-server for a customer, its really pain in the ass.

Another thing, I can see where it is microservice friendly and where it isnt, but its really too early for that.

Rather then that, Im here to stay.
Thanks.

**Comment by yashraj2003e** [comment]
I have been part of the community for a long time and that is something great. Congratulations 👏.

**Comment by Karalix** [comment]
Congratulations on the launch! I had a great experience thus far, with only a slight frustration. I spent a lot of time looking for details of how Accounts worked. A more developed documentation on this would have helped a lot.

**Reply by Karalix** [reply]
To be more precise, I think it was not explicit enough what one should keep in the store of the app to keep in memory the auth state.

**Comment by jyotirmaya907** [comment]
Hello, everyone!! 
My name is Jyotirmaya Jena
In last month I joined this community which is be always active. I know about Appwrite and also used in some project also. It's was easy to use , no that much of complex. As for now it was nice experience with it.

**Comment by dcowan-london** [comment]
Before AppWrite, I'd never used a BaaS (unless you count Strapi, which I don't - it does the job it's designed to do, which is a CMS backend). I can't compare to anything else, because I've never used anything else.

I head about AppWrite on Reddit, and thought "that looks cool". I installed it to have a look round, and before I knew it, it had become an indispensable part of a project I was working on.

The transition from traditional backends has been incredible. It's taken away a huge block in starting a new small project. And I can of course still have my own more custom backend with Functions.

The documentation is incredible, and the community (on Discord) is really great and always helpful.

I've got two suggestions - the first one is global user management - #4044!

The second is to have rate limits (https://appwrite.io/docs/rate-limits) be configurable to be per project instead of per instance - and to be able to disable / reenable rate limits without needing to restart AppWrite.

You seem to be making a huge amount of progress - and I think Cloud will be a big step too! I'm also looking forward to Hacktober!

Congratulations on the launch and on Product Hunt #1!

**Comment by trishitapingolia** [comment]
Congratulations 🎉 on the launch to team AppWrite. Hope the new features make great improvements. 
I had a great experience with AppWrite till now and I am proud to be a member of the community. ❤️

**Comment by rumbis** [comment]
AppWrite is lifechanging recently I found it  and I like a lot  that you got everything  as simple as possible  still researching it how to have a better use for now on and if I compare with other solutions around  is  the greatest  tool and its only at v1 yet . thanks for your effort and looking forward how I can contribute on that amazing service, product and data magic.

**Comment by vini-brito** [comment]
Would be great to have a GET endpoint as an entry point to a web app. Right now I have to keep the entry point to a web app in a separate host. I mean, something I could place in my DNS records and it would serve the page's HTML.

**Comment by ChauCM** [comment]
Love AppWrite for my Flutter app. Being able to write Dart cloud function literally won the 'base competition for me (guess what, Firebase doesn't support their programming language Dart for cloud function). However, I'm not a fan of the way we need to getAccount to check for authentication. I think Firebase is doing a really good job at it by providing you subscriptions like onAuthChanged. Supabase 1.0 also moving to onAuthStateChange. Really hope this will be implemented in the future.
